specification/billing/resource-manager/Microsoft.Billing/stable/2021-10-01/payment.json

```diff
@@ -0,0 +1,883 @@
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/billingAccountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/billingProfileNameParameter"
+          },
+          {
+            "$ref": "#/parameters/paymentMethodNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/PaymentMethodLink"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "PaymentMethods"
+        ],
+        "operationId": "PaymentMethodAtBillingProfile_Remove",
+        "x-ms-long-running-operation": true,
+        "description": "Deletes a payment method link and removes the payment method from a billing profile. The operation is supported only for billing accounts with agreement type Microsoft Customer Agreement.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/billing/"
+        },
+        "x-ms-examples": {
+          "PaymentMethodAtBillingProfileRemove": {
+            "$ref": "./examples/PaymentMethodAtBillingProfile_Remove.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/billingAccountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/billingProfileNameParameter"
+          },
+          {
+            "$ref": "#/parameters/paymentMethodNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded."
+          },
+          "204": {
+            "description": "The resource was not found."
+          },
+          "202": {
+            "description": "Accepted",
+            "headers": {
+              "Location": {
+                "description": "Location URI to poll for result",
+                "type": "string"
+              },
+              "Retry-After": {
+                "description": "Recommended time to wait before making another request to check the status of the operation. The time is specified in seconds.",
+                "type": "integer",
+                "format": "int32"
+              }
+            }
+          },
+          "409": {
+            "description": "Conflict. Payment method cannot be detached from billing profile.",
+            "x-ms-error-response": true,
+            "schema": {
+              "$ref": "#/definitions/DetachPaymentMethodEligibilityResult"
```

Comment:
>definitions/DetachPaymentMethodEligibilityResult" [](http://example.com/codeflow?start=25&length=49)

This should be the ARM error schema (definition/ErrorResponse). Actually you don't really need to specify 409 here since it will fall into "default".

Reply:
There are specific error codes enum for this scenario. Don't we have to mention those as part of new schema?

Reply:
modified the schema to mirror ARM error schema

Reply:
Not sure how this would impact SDK generation. I will let the SDK team review.